### DISCUSSION TITLE
`og:image` fails to load for LinkedIn, Threads but works on Twitter, iMessage

### DISCUSSION THREAD

**Discussion by libhide** [vercel/next.js#60180] [Help] [answered]
### Summary

Hey! 👋🏼 

I am using Next.js 13 with the app router for my personal website and running into a rather peculiar problem. I'm on the latest canary version (since I wanted to use partial prerendering). The problem I'm seeing is with social media share images failing to load on _certain_ websites.

If I try to share `https://www.ratik.in/writing/default-apps-2023` (a blog post on my website) to Twitter or send it as a link on iMessage, the share image loads in just fine. But, if I try to share the same link on LinkedIn or Thread, a preview fails to be generated. 

I'm attaching two screenshots showing exactly what happens when I try to share a link to a blog post: (1) the error LinkedIn gives me, (2) Runtime logs from Vercel. I'm also pasting in the code for the `writing/[slug].tsx` page in case that helps get to the bottom of this.

Hope someone can help me get to the bottom of what's happening here.

Cheers!

![LinkedIn Error](https://github.com/vercel/next.js/assets/1370559/64506f76-5bbc-437d-aa86-36da879bfcc4)

![LinkedIn Log](https://github.com/vercel/next.js/assets/1370559/4638b21f-b47a-4ac7-8ab5-98de84dc3e57)

### Additional information

```js
import type { Metadata, ResolvingMetadata } from "next"

import { getPostByFileName, getPostPreviews } from "@/lib/writing"

type PageProps = {
  params: {
    slug: string
  }
}

export async function generateMetadata(
  { params: { slug } }: PageProps,
  parent: ResolvingMetadata,
): Promise<Metadata> {
  const post = await getPostByFileName(`${slug}.mdx`)
  const parentMetadata = await parent

  const title = post.meta.title
  const description = post.meta.description

  const parentOpenGraph = parentMetadata.openGraph

  return {
    title: title,
    description: description,
    keywords: post.meta.tags,
    openGraph: {
      ...parentOpenGraph,
      title: title,
      description: description,
      url: `/writing/${post.meta.slug}`,
      images: [
        {
          url: `${post.meta.seoImg}`,
        },
      ],
    },
    twitter: {
      title: title,
      description: description,
      images: [`${post.meta.seoImg}`],
    },
  }
}

export async function generateStaticParams() {
  const posts = await getPostPreviews()
  return posts.map((post) => ({
    slug: post.slug,
  }))
}

const PostPage = async ({ params: { slug } }: PageProps) => {
  const post = await getPostByFileName(`${slug}.mdx`)

  return (
    <div className="m-auto max-w-2xl lg:max-w-3xl">
      <section className="px-6 lg:px-16">
        <div className="mt-6 lg:mt-16 2xl:mt-24">
          <h1 className="mb-4 text-3xl font-bold lg:text-4xl lg:text-5xl 2xl:text-6xl">
            {post.meta.title}
          </h1>
          <p className="mb-8 text-gray-500">Published: {post.meta.date}</p>
          <article id="post" className="">
            {post.content}
          </article>
        </div>
      </section>
    </div>
  )
}

export default PostPage
```

**Comment by derek-cook** [comment]
@libhide Did you ever get this to work? I've had the same issue but the og image shows up everywhere _except_ iMessage

**Comment by libhide** [comment]
@derek-cook Nope, but there is an issue related to this now: [issue](https://github.com/vercel/next.js/issues/60620). Hopefully, there will be a fix soon. 🤞🏼

**Comment by libhide** [comment, marked as answer]
This issue has been fixed: https://github.com/vercel/next.js/issues/60620
